crux-mir/lib/libstd/sys/crux/mutex.rs

```diff
@@ -22,14 +24,16 @@ impl Mutex {
     pub unsafe fn lock(&self) {
         // TODO: Currently, we never run `drop` code, so locks are never released.  Once we support
         // drops, we can enable this assertion to check for (invalid) reentrant locking.
-        //assert!(!self.locked.get());
-        self.locked.set(true);
+        // assert!(!self.locked.get());
+        concurrency::mutex_lock(self);
+        self.locked.set(true); //TODO: Why does this yield an empty mux tree in crux?
```

Comment:
Is this TODO still relevant?  If so, under what conditions does it give the "empty mux tree" error?  (I assume it's not always doing it since otherwise nothing would work.)

Reply:
Agreed: I think I confused myself when rebasing and thought that this was actually a pre-existing comment somehow, referring to something that I wasn't observing. While this PR _is_ experimental, it's not so experimental that it's crashing on every run.